generators/beamAndVertexGenerator.cc

```diff
@@ -79,21 +121,62 @@ bool beamAndVertexGenerator::loadBeamFile(const string& beamFileName)
 	if(not (vertexXPositionSigma and vertexYPositionSigma and
 	        beamMomentumXSigma and beamMomentumYSigma and beamMomentumZSigma))
 	{
-		printWarn << "One of the optional branches is missing in tree '" << beamTreeName
+		printWarn << "One of the optional branches is missing in tree '" << _beamTree->GetName()
 		          << "' in root file '" << _beamFileName << ". Required are"
```

Comment:
Warning message is a bit vague: "optional branches" contradicts somewhat "Required"; it would be good to state which branches would be needed to do what.